proctor-codegen/src/main/resources/com/indeed/proctor/consumer/ant/groups.ftl

```diff
@@ -184,6 +187,27 @@ public class ${mainClassName} extends AbstractGroups {
         return payload.${testDef.payloadAccessorName}();
     }
     </#if>
+
+    <#if (testDef.isJson)??>
+    public @Nullable <T> T get${testDef.javaClassName}Payload(final Class<T> payloadType) {
+        final Payload payload = getPayload(${testEnumName}.${testDef.enumName}.getName(), ${testEnumName}.${testDef.enumName}.getFallbackValue());
+        ObjectMapper mapper = new ObjectMapper();
+
+        if (payload == null) {
+            return null;
+        }
+
+        T payloadCastedToType = null;
+
+        try {
+            payloadCastedToType = mapper.treeToValue(payload.${testDef.payloadAccessorName}(), payloadType);
+        } catch (JsonProcessingException e) {
+            return null;
```

Comment:
I wonder if we should throw `JsonProcessingException` here (adding `throws JsonProcessingException` signature to the method too) so that users can handle it in their way (e.g., they can log ERROR message to notify something wrong in payload value). However, it also makes this method less easy to use. (e.g., every places needs `try { ... } catch { ... }`), so I'm not sure which is better.

@isozuka @taitori @zacharygoodwin Any preferences between the two options?

Reply:
I don't have any preference, however one thing to note is that they still would also have to check for null in the case that a payload does not exist. So could be current implementation which is just return null, or return null and throw JsonProcessing Exception, or another option would be to throw both JsonProcessingException and NullPointerException.

Reply:
> one thing to note is that they still would also have to check for null in the case that a payload does not exist.

That's a good point. Considering that point, the current way (just return null) looks better to me for simplicity.